### DISCUSSION TITLE
tor.streamisolation — how is it different from Tor's own stream isolation?

### DISCUSSION THREAD

**Discussion by xanoni** [lightningnetwork/lnd#5783] [Q&A] [answered]
```
; Enable Tor stream isolation by randomizing user credentials for each
; connection. With this mode active, each connection will use a new circuit.
; This means that multiple applications (other than lnd) using Tor won't be mixed
; in with lnd's traffic.

tor.streamisolation=true
```

Let's say I have [set up SOCKSPORT as follows](http://jqyzxhjk6psc6ul5jnfwloamhtyh7si74b4743k2qgpskwwxrzhsxmad.onion/docs/tor-manual.html.en):
```
SocksPort 8075 IsolateClientProtocol IsolateDestPort IsolateDestAddr ExtendedErrors IPv6Traffic PreferIPv6 KeepAliveIsolateSOCKSAuth
```

Does `tor.streamisolation=true` add any benefit here, or is it duplicative?

**Comment by guggero** [comment, marked as answer]
I looked at the code of [what lnd does with the `tor.streamisolation` flag](https://github.com/lightningnetwork/lnd/blob/master/tor/tor.go#L110) and it looks like it creates a new set of credentials for each individual connection which in turn causes Tor to create a new circuit every time an address is dialed.
So this seems to create an additional layer of isolation. But looking at what `IsolateDestAddr` does and how "expensive" a new circuit is, it might be a good middle ground to use `IsolateDestAddr` instead of `lnd`'s stream isolation. That way you could re-use an existing circuit to re-connect to a node (if it just restarts) instead of forcing a new circuit on every connection attempt.
That _could_ lead to somewhat more stable connections, but I'm just speculating here.

**Reply by xanoni** [reply]
This is what I was hoping to hear. I've read that people have had issues with `tor.streamisolation`, see https://github.com/rootzoll/raspiblitz/issues/2560.

Personally, my stability issues got much better after I launched a dedicated Tor instance just for `lnd` ... however, I'm now also disabling this flag, given there is no risk of cross-pollution at all if only `lnd` is using the SOCKSPORT.
